### PR TITLE
Move some scripts to framework submodule

### PR DESCRIPTION
**Pull request by davidhorstmann-arm**
Move `code_style.py` and test code generation scripts to the framework submodule.

Depends on Mbed-TLS/mbedtls-framework#5

Note: The `framework` submodule is updated to the tip of Mbed-TLS/mbedtls-framework#5 to get the CI results. I can rebase this once that PR is merged.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required - internal change only
- [x] **3.6 backport** #9012 
- [x] **2.28 backport** not required - framework is 3.x only
- [x] **tests** not required file moves only

### CODE REVIEW

**Comment by davidhorstmann-arm** [comment]
[Force-pushed to update to version of submodule with full history]

**Review by Ryan-Everett-arm — CHANGES_REQUESTED** [review]
I have found two comments in the codebase which still reference the old paths (sorry for not leaving my review comments on those lines - I don't think Github lets you comment on unchanged files). I am happy with everything else, though!

**Comment by davidhorstmann-arm** [comment]
Thanks for catching these, should be addressed now!

**Review by Ryan-Everett-arm — APPROVED** [review]
LGTM, thanks!

**Review by Ryan-Everett-arm — CHANGES_REQUESTED** [review]
There are some CI failures in the trivial 3.6 backport (see my comment [here](https://github.com/Mbed-TLS/mbedtls/pull/9012#pullrequestreview-1980633899)). I think that this branch needs to be rebased to pull in the addition of  [tests/scripts/generate_psa_wrappers.py](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/scripts/generate_psa_wrappers.py) so that we can change the references to `mbedtls_dev` in this file, otherwise the merge CI will likely fail.

**Review by Ryan-Everett-arm — CHANGES_REQUESTED** [review]
Thanks for the changes. It seems that there are now two scripts in both `mbedtls_dev` and `framework_dev`, everything else looks good though!

**Review by Ryan-Everett-arm — APPROVED** [review]
LGTM, thanks!

File: tests/scripts/check-python-files.sh [hunk 1]
```diff
@@ -55,14 +55,14 @@ elif [ "$1" = "--can-mypy" ]; then
 fi
 
 echo 'Running pylint ...'
-$PYTHON -m pylint scripts/mbedtls_dev/*.py scripts/*.py tests/scripts/*.py || {
+$PYTHON -m pylint framework/scripts/framework_dev/*.py scripts/*.py tests/scripts/*.py || {
```
Comment (ronald-cron-arm) [inline tests/scripts/check-python-files.sh:58]: Like in line 65, why not adding framework/scripts/*.py here as well?

File: tests/scripts/check_files.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
     pass
 
 import scripts_path # pylint: disable=unused-import
-from mbedtls_dev import build_tree
```
Comment (ronald-cron-arm) [inline tests/scripts/check_files.py:27]: This file has to be modified to check the files that are in the `framework` submodule as well. It seems we just have to change `bytes_output = subprocess.check_output(['git', 'ls-files', '-z'])` to `bytes_output = subprocess.check_output(['git', 'ls-files', '--recurse-submodules', '-z'])`.

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]

**Comment by davidhorstmann-arm** [comment]
Closing as superseded by more separate PRs

**Pull request closed without merging**